Overriding equals must also override hashcode,

cameroncros opened the pull request:
Position classes are now immutable
PositionGPS.compare correctly returns directional information as well
Tests for positionMetric added
Tests reorganised

Readded some things that were removed but really should stay. Code is sticking to java's best practices.

cameroncros committed:
Overriding equals must also override hashcode,
Position classes are now immutable
PositionGPS.compare correctly returns directional information as well
Tests for positionMetric added
Tests reorganised

7178301 commented:
Position classes should be declared final and their attributes for immutability
PositionMetric.java equals function -> latitude should be before longitude
PositionGPS.add -> shouldn't it return PositionGPS not Position
Code reference on line 33 is invalid. Code generated using Great Circle Distance Mathematical Equation

cameroncros committed:
Position/Angle classes are now final
Fixed order of latitude and longitude in equals
Position.add functions now return correct type in subclasses

cameroncros committed:
Readded comment attributing code to Apache SIS
